base/strings/annotated.jl

```diff
@@ -125,7 +126,10 @@ AnnotatedString(s::AnnotatedString, annots::Vector{Tuple{UnitRange{Int}, Pair{Sy
 AnnotatedChar(c::AnnotatedChar, annots::Vector{Pair{Symbol, Any}}) =
     AnnotatedChar(c.char, vcat(c.annotations, annots))
 
-String(s::AnnotatedString{String}) = s.string # To avoid pointless overhead
+# To allow for generically de-annotating a string.
+(::Type{T})(s::AnnotatedString{T}) where {T <: AbstractString} = T(s.string)
```

Comment:
So this is genuinely ambiguous with respect to at least these methods, possibly more:
```
GenericString(string::AbstractString) @ Base strings/annotated.jl:130
SubString(s::AbstractString) @ Base strings/substring.jl:59
Core.Compiler.LazyString(args...) @ Core.Compiler strings/lazy.jl:41
```

I am not sure it is possible to resolve this (esp. if one also takes the potential for similar such issues with packages into account). Not am I sure it is necessary: sure instead of `MyStringType(my_annotate_str)` it is also OK if I have to write `MyStringType(String(my_annotate_str))` ? The necessary method is already in `master`